projects/packages/jetpack-mu-wpcom/src/features/wpcom-sidebar-notice/wpcom-sidebar-notice.js

```diff
@@ -15,50 +15,15 @@ const wpcomSidebarNoticeRecordEvent = ( event, wpcomSidebarNoticeData ) => {
 	);
 };
 
-const wpcomShowSidebarNotice = wpcomSidebarNoticeData => {
-	const adminMenu = document.querySelector( '#adminmenu' );
-	if ( ! adminMenu || ! wpcomSidebarNoticeData ) {
+const wpcomShowSidebarNotice = () => {
+	const sidebarNotice = document.querySelector( '#toplevel_page_site-notices' );
+	if ( ! sidebarNotice || ! wpcomSidebarNoticeData ) {
 		return;
 	}
 
-	// Render notice.
-	adminMenu.insertAdjacentHTML(
-		'afterbegin',
-		`
-			<li
-				id="toplevel_page_site-notices"
-				class="wp-not-current-submenu menu-top menu-icon-generic toplevel_page_site-notices"
-				data-id="${ wpcomSidebarNoticeData.id }"
-				data-feature-class="${ wpcomSidebarNoticeData.featureClass }"
-			>
-				<a href="${
-					wpcomSidebarNoticeData.url
-				}" class="wp-not-current-submenu menu-top menu-icon-generic toplevel_page_site-notices">
-					<div class="wp-menu-name">
-						<div class="upsell_banner">
-							<div class="upsell_banner__icon dashicons" aria-hidden="true"></div>
-							<div class="upsell_banner__text">${ wpcomSidebarNoticeData.text }</div>
-							<button type="button" class="upsell_banner__action button">${
-								wpcomSidebarNoticeData.action
-							}</button>
-							${
-								wpcomSidebarNoticeData.dismissible
-									? '<button type="button" class="upsell_banner__dismiss button button-link">' +
-									  wpcomSidebarNoticeData.dismissLabel +
-									  '</button>'
-									: ''
-							}
-						</div>
-					</div>
-				</a>
-			</li>
-		`
-	);
-
 	// Record impression event in Tracks.
-	wpcomSidebarNoticeRecordEvent( wpcomSidebarNoticeData.tracks?.display );
+	wpcomSidebarNoticeRecordEvent( wpcomSidebarNoticeData.tracks?.display, wpcomSidebarNoticeData );
```

Comment:
Note that the second arg was missing here, causing an error in the function, and causing everything after this line to not be executed.